src/c++/perf_analyzer/genai-perf/genai_perf/llm_metrics.py

```diff
@@ -627,18 +627,14 @@ def _preprocess_response(
                     res_outputs[i] = {"response": json.dumps(merged_response)}
 
             # Remove responses without any content
-            # These are only observed to happen at the start or end
-            while res_outputs and self._is_openai_empty_response(
-                res_outputs[0]["response"]
-            ):
-                res_timestamps.pop(0)
-                res_outputs.pop(0)
-
-            while res_outputs and self._is_openai_empty_response(
-                res_outputs[-1]["response"]
-            ):
-                res_timestamps.pop()
-                res_outputs.pop()
+            indices_to_remove = []
+            for idx, out in enumerate(res_outputs):
+                if self._is_openai_empty_response(out["response"]):
```

Comment:
do we want to limit this to just openai?
is this not possible in any other set ups?

Reply:
I think it would require some investigation into the others, it might be some strange http artifact.  I think we could follow it up, but atm this small change is necessary.